DashAI/front/src/components/datasets/DatasetModal.jsx

```diff
@@ -64,6 +64,7 @@ function DatasetModal({ open, setOpen, updateDatasets }) {
       formData.append("file", newDataset.file);
       await uploadDatasetRequest(formData);
       enqueueSnackbar("Dataset uploaded successfully", { variant: "success" });
+      updateDatasets();
```

Comment:
This function is to fetch the datasets?
Maybe if it does that, then it would be good to put getDatasets instead of updateDatasets.


If so, what happens if this call fails?
Maybe there should be an exclusive snackbar for this?

It could be that you can pull out which call failed from the errors to handle them by error type.

Reply:
I'm seeing that the states are getting pretty nested (I saw about 3 components before I knew where the datasets were).

I think it's a good time to consider having centralized states with redux or something like that.
Would you be interested in discussing it?